scene/main/node.cpp

```diff
@@ -2091,74 +2092,6 @@ Node *Node::duplicate_from_editor(Map<const Node *, Node *> &r_duplimap) const {
 }
 #endif
 
-void Node::_duplicate_and_reown(Node *p_new_parent, const Map<Node *, Node *> &p_reown_map) const {
-
-	if (get_owner() != get_parent()->get_owner())
-		return;
-
-	Node *node = NULL;
-
-	if (get_filename() != "") {
-
-		Ref<PackedScene> res = ResourceLoader::load(get_filename());
-		ERR_FAIL_COND(res.is_null());
-		node = res->instance();
-		ERR_FAIL_COND(!node);
-	} else {
-
-		Object *obj = ClassDB::instance(get_class());
-		ERR_EXPLAIN("Node: Could not duplicate: " + String(get_class()));
-		ERR_FAIL_COND(!obj);
-		node = Object::cast_to<Node>(obj);
-		if (!node)
-			memdelete(obj);
```

Comment:
That refactoring fixes at least one bug: `ERR_FAIL_COND_V(!node, NULL);` was missing here, so it could dereference a null pointer below.